beacon_node/eth2-libp2p/src/rpc/methods.rs

```diff
@@ -219,9 +149,27 @@ pub struct BeaconBlockBodiesRequest {
 }
 
 /// Response containing the list of requested beacon block bodies.
-#[derive(Encode, Decode, Clone, Debug, PartialEq)]
+#[derive(Clone, Debug, PartialEq)]
 pub struct BeaconBlockBodiesResponse {
-    /// The list of beacon block bodies being requested.
+    /// The list of hashes that were sent in the request and match these roots response. None when
+    /// sending outbound.
+    pub block_roots: Option<Vec<Hash256>>,
+    /// The list of ssz-encoded beacon block bodies being requested.
+    pub block_bodies: Vec<u8>,
+}
+
+/// The decoded version of `BeaconBlockBodiesResponse` which is expected in `SimpleSync`.
+pub struct DecodedBeaconBlockBodiesResponse {
```

Comment:
The current logic is. RPC -> Message Handler (Decodes and creates this object: DecodedBeaconBlockBodiesResponse) -> SimpleSync. 

In the encoding/decoding, currently use `EncodeableBeaconBlockBodiesResponse` below. Suggestions for a better way of doing this without creating 3 types? 
I need a third type, because the `block_roots` isn't sent across the wire. I just keep track of it and send it back in the response for utility

Reply:
To ensure I understand, are these the tree types?

- Request
- Response
- Response w/ Request roots